File: week-2/week2-3.py
def maxProduct(nums):
    num = len(nums)
    r1,r2=[],[]
    for i in range(num):
        for j in nums:
            mul=nums[i]*j
            r1.append(mul)
        r2.append(nums[i]**2)
    # print(r1)
    # print(r2)
    for x in r2:
        r1.remove(x)
    max_ans=max(r1)
    print(max_ans)

maxProduct([5, 20, 2, 6])
maxProduct([10, -20, 0, 3])
maxProduct([-1, 2])
maxProduct([-1, 0, 2])
maxProduct([-1, -2, 0])

File: week-2/week2-3.js
function maxProduct(nums){
    let num = nums.length
    let r1=[]
    for(let i=0; i<num; i++){
        for(let j=0; j<num; j++){
            if(i!==j){
                let mul=nums[i]*nums[j]
                r1.push(mul)
            }
        }
    }
    // console.log(r1)
    let ans=Math.max(...r1)
    console.log(ans)
}
maxProduct([5, 20, 2, 6])
maxProduct([10, -20, 0, 3])
maxProduct([-1, 2])
maxProduct([-1, 0, 2])
maxProduct([-1, -2, 0])